lgsm/functions/workshop_core.sh

```diff
@@ -0,0 +1,318 @@
+		if [ -d "${workshopmodsdir}/${modid}" ]; then
+			rm -rf "${workshopmodsdir}/${modid}"
+		fi
+		modsrc="${workshopmodsdldir}/steamapps/workshop/content/${gameappid}/${modid}"
+		cp -fa ${modsrc} ${workshopmodsdir}
+		if [ "${engine}" == "realvirtuality" ]; then
+			modkey="${workshopmodsdldir}/steamapps/workshop/content/${gameappid}/${modid}/keys"
+			if ! [ -d "${modkey}" ]; then
+				modkey="$steamcmd/steamapps/workshop/content/${gameappid}/${modid}/Keys"
+			fi
+			if ! [ -d "${modkey}" ]; then
+				modkey="$steamcmd/steamapps/workshop/content/${gameappid}/${modid}/key"
+			fi
+			if ! [ -d "${modkey}" ]; then
+				modkey="$steamcmd/steamapps/workshop/content/${gameappid}/${modid}/Key"
+			fi
+			if ! [ -d "${modkey}" ]; then
+				echo "Mod ${modname} seems to be missing key folder. Tring to copy key from the main folder."
+				cp -fa "${workshopmodsdir}/${modid}/*.bikey" ${keysdir}
+			else
+				cp -fa ${modkey}/*.bikey ${keysdir}
+			fi
+		fi
+	else
+		echo "Mod ${modname}  is already in mods folder."
+	fi
+}
+
+# ## Directory management.
+
+# Create mods files and directories if it doesn't exist.
+fn_create_workshop_dir() {
+	# Create lgsm data modsdir.
+	if [ ! -d "${workshopmodsdldir}" ]; then
+		echo -en "creating LinuxGSM Steam Workshop data directory ${workshopmodsdldir}..."
+		mkdir -p "${workshopmodsdldir}"
+		exitcode=$?
+		if [ "${exitcode}" != 0 ]; then
+			fn_print_fail_eol_nl
+			fn_script_log_fatal "Creating mod download dir ${workshopmodsdldir}"
+			core_exit.sh
+		else
+			fn_print_ok_eol_nl
+			fn_script_log_pass "Creating mod download dir ${workshopmodsdldir}"
+		fi
+	fi
+	# Create mod install directory.
+	if [ ! -d "${workshopmodsdir}" ]; then
+		echo -en "creating Steam Workshop install directory ${workshopmodsdir}..."
+		mkdir -p "${workshopmodsdir}"
+		exitcode=$?
+		if [ "${exitcode}" != 0 ]; then
+			fn_print_fail_eol_nl
+			fn_script_log_fatal "Creating mod install directory ${workshopmodsdir}"
+			core_exit.sh
+		else
+			fn_print_ok_eol_nl
+			fn_script_log_pass "Creating mod install directory ${workshopmodsdir}"
+		fi
+	fi
+}
+
+# Counts how many mods were installed.
+fn_workshop_count_installed() {
+	if [ -f "${workshopmodsdir}" ]; then
+		installedworkshopmodscount=$(ls -l "${workshopmodsdir}" | grep -c ^d)
+	else
+		installedworkshopmodscount=0
+	fi
+}
+
+# Exits if no mods were installed.
+fn_workshop_check_installed() {
+	# Count installed mods.
+	fn_workshop_count_installed
+	# If no mods are found.
+	if [ ${installedworkshopmodscount}/* -eq 0 ]; then
+		echo -e ""
+		fn_print_failure_nl "No installed workshop mods or addons were found"
+		echo -e " * Install mods using LinuxGSM first with: ./${selfname} workshop-install"
+		fn_script_log_error "No installed workshop mods or addons were found."
+		core_exit.sh
+	fi
+}
+
+# Builds list of installed Steam Workshop mods.
+fn_workshop_installed_list() {
+	fn_workshop_count_installed
+	for folder in ${workshopmodsdir}/*; do
```

Comment:
Unprotected variable, although there are a few of these. I'm not one of those people who insists on everything being quote-protected, because not everything needs to be, such as integers. However, since it's a directory, one which is presumably changeable by the user or a developer, it's always best to quote-protect it. Unprotected variables like paths can lead to destructive results.